bin/sync_slurm_tier1_projects.py

```diff
@@ -0,0 +1,200 @@
+    slurm_project_accounts, slurm_project_users_accounts,
+    slurm_project_qos
+)
+from vsc.utils.py2vs3 import HTTPError
+from vsc.utils.run import RunNoShell
+
+VSC_ADMIN_GROUPS = ("astaff", "badmin", "l_sysadmin", "gt1_dodrio_vscadmins")
+
+TIER1_PROTECTED_ACCOUNTS = ("root", "projects", "gadminforever", "gt1_default")
+
+def execute_commands(commands):
+    """Run the specified commands"""
+
+    for command in commands:
+        logging.info("Running command: %s", command)
+
+        # if one fails, we simply fail the script and should get notified
+        (ec, _) = RunNoShell.run(command)
+        if ec != 0:
+            raise SacctMgrException("Command failed: {0}".format(command))
+
+
+ProjectIniConfig = namedtuple("ProjectIniConfig",
+    ["name", "group", "end_date", "members", "cpu_hours", "gpu_hours"]
+)
+
+
+def over_and_done(datestamp):
+    """
+    Is the datestamp past the current day?
+    """
+    end = datetime.strptime(datestamp, "%Y%m%d").date()
+    return end < datetime.today().date()
+
+
+def get_projects(projects_ini):
+    """
+    This reads the ini file which contains information on the projects
+
+    [project_name]
+    end_date: YYYYMMDD
+    members: comma-separated list of VSC IDs of project members
+    CPUhours: int
+    GPUhours: int
+
+    """
+
+    projects_config = ConfigParser()
+    with open(projects_ini) as pini:
+        projects_config.read_file(pini)
+
+    active_projects = []
+    past_projects = []
+
+    for section in projects_config.sections():
+        if section in TIER1_PROTECTED_ACCOUNTS:
+            continue
+
+        end_date = projects_config.get(section, "end_date", fallback="20380101")
+
+        if over_and_done(end_date):
+            logging.debug("Project %s past end date %s", section, end_date)
+            projects = past_projects
+        else:
+            projects = active_projects
+
+        logging.info("processing section %s", section)
+        if section in VSC_ADMIN_GROUPS:
+            projects.append(ProjectIniConfig(
+                name=section,
+                group=section,
+                end_date=projects_config.get(section, "end_date"),
+                members=[m.strip() for m in projects_config.get(section, "members").split(",")],
+                cpu_hours=int(projects_config.get(section, "CPUhours", fallback=0)),
+                gpu_hours=int(projects_config.get(section, "GPUhours", fallback=0)),
+            ))
+
+        elif section.startswith('gpr_compute'):
+            projects.append(ProjectIniConfig(
+                name=section.replace("gpr_compute_", ""),
+                group=section,
+                end_date=projects_config.get(section, "end_date"),
+                members=[m.strip() for m in projects_config.get(section, "members").split(",")],
+                cpu_hours=int(projects_config.get(section, "CPUhours", fallback=0)),
+                gpu_hours=int(projects_config.get(section, "GPUhours", fallback=0)),
+            ))
+
```

Comment:
as usual, at least log the else; and if we don't expect anything not matching, fail the sync